core/services/src/main/java/org/exoplatform/services/attachments/service/AttachmentServiceImpl.java

```diff
@@ -282,6 +301,40 @@ public Attachment getAttachmentById(String attachmentId, SessionProvider session
     }
   }
 
+  @Override
+  public void moveAttachmentToNewPath(long userIdentityId,
+                                      String attachmentId,
+                                      String newPathDrive,
+                                      String newPath) throws Exception {
+    if (userIdentityId <= 0) {
+      throw new IllegalAccessException("User identity must be positive");
+    }
+    if (StringUtils.isEmpty(attachmentId)) {
+      throw new IllegalArgumentException("Attachment id is mandatory");
+    }
+    if (StringUtils.isEmpty(newPathDrive)) {
+      throw new IllegalArgumentException("New destination path's drive must not be empty");
+    }
+
+    SessionProvider sessionProvider = sessionProviderService.getSystemSessionProvider(null);
+    String workspace = repositoryService.getCurrentRepository().getConfiguration().getDefaultWorkspaceName();
+    Session session = sessionProvider.getSession(workspace, repositoryService.getCurrentRepository());
+    try {
+      Node attachmentNode = session.getNodeByUUID(attachmentId);
+      Node destNode = getParentFolderNode(workspace, session, newPathDrive, newPath);
+      String destPath = destNode.getPath().concat("/").concat(attachmentNode.getName());
+      session.move(attachmentNode.getPath(), destPath);
+      attachmentNode.refresh(true);
```

Comment:
why refresing, isn't the `session.save()` sufficient ?

Reply:
yes it is sufficient.